### PR TITLE
OBSDOCS-996: Create a table listing and describing all RBAC permissio…

### PR DESCRIPTION
**Pull request by eromanova97**
Version(s): `enterprise-4.12` and later

Issue: [OBSDOCS-996](https://issues.redhat.com/browse/OBSDOCS-996)

Link to docs preview:  [Granting users permission to monitor user-defined projects
](https://75852--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/monitoring/enabling-monitoring-for-user-defined-projects.html#granting-users-permission-to-monitor-user-defined-projects_enabling-monitoring-for-user-defined-projects)

QE review:
- [ ] QE has approved this change.

**Additional information:**

### CODE REVIEW

File: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc [hunk 1]
```diff
@@ -6,22 +6,28 @@
...
 
-* The *monitoring-rules-edit* cluster role grants a user permission to create, modify, and delete `PrometheusRule` custom resources for a project. It also grants a user the ability to silence alerts.
+You can grant the permissions by assigning one of the following monitoring roles:
 
-* The *monitoring-edit* cluster role grants the same privileges as the `monitoring-rules-edit` cluster role. Additionally, it enables a user to create new scrape targets for services or pods. With this role, you can also create, modify, and delete `ServiceMonitor` and `PodMonitor` resources.
+|===
+|Cluster role name |Description
 
-You can also grant users permission to configure the components that are responsible for monitoring user-defined projects:
+|`monitoring-rules-view` | Users with this role have read access to `PrometheusRule` custom resources for a user-defined project.
```
[the remark this answers is not in the document]
Comment (eromanova97) [reply modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc:22]: That would be definitely helpful to mention, thank you!

File: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc [hunk 2]
```diff
@@ -6,22 +6,28 @@
...
+|===
+|Cluster role name |Description
 
-You can also grant users permission to configure the components that are responsible for monitoring user-defined projects:
+|`monitoring-rules-view` | Users with this role have read access to `PrometheusRule` custom resources for a user-defined project. They can also view the *Alerts* page in the *Developer* perspective of the {product-title} web console.
 
-* The *user-workload-monitoring-config-edit* role in the `openshift-user-workload-monitoring` project enables you to edit the `user-workload-monitoring-config` `ConfigMap` object. With this role, you can edit the `ConfigMap` object to configure Prometheus, Prometheus Operator, and Thanos Ruler for user-defined workload monitoring.
+|`monitoring-rules-edit` | Users with this role can create, modify, and delete `PrometheusRule` custom resources for a user-defined project. They can also silence alerts.
```
Comment (juzhao) [inline modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc:24]:
```suggestion
|`monitoring-rules-edit` | Users with this role can create, modify, and delete `PrometheusRule` custom resources for a user-defined project. They can create and silence alerts in the `Developer` perspective in the web console
```
see from: https://docs.openshift.com/container-platform/4.15/observability/monitoring/managing-alerts.html

File: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc [hunk 3]
```diff
@@ -6,22 +6,28 @@
...
 
-* The *monitoring-rules-edit* cluster role grants a user permission to create, modify, and delete `PrometheusRule` custom resources for a project. It also grants a user the ability to silence alerts.
+You can grant the permissions by assigning one of the following monitoring roles:
 
-* The *monitoring-edit* cluster role grants the same privileges as the `monitoring-rules-edit` cluster role. Additionally, it enables a user to create new scrape targets for services or pods. With this role, you can also create, modify, and delete `ServiceMonitor` and `PodMonitor` resources.
+|===
+|Cluster role name |Description
 
-You can also grant users permission to configure the components that are responsible for monitoring user-defined projects:
+|`monitoring-rules-view` | Users with this role have read access to `PrometheusRule` custom resources for a user-defined project. They can also view the *Alerts* page in the *Developer* perspective of the {product-title} web console.
```
Comment (juzhao) [inline modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc:22]:
```suggestion
|`monitoring-rules-view` | Users with this cluster role have read access to `PrometheusRule` custom resources for a user-defined project. They can also view the *Alerts* page in the *Developer* perspective of the {product-title} web console.
```
reason
```
$ oc get clusterrole monitoring-rules-view
NAME                    CREATED AT
monitoring-rules-view   2024-05-14T03:41:36Z
```

File: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc [hunk 4]
```diff
@@ -6,22 +6,28 @@
...
 
-* The *user-workload-monitoring-config-edit* role in the `openshift-user-workload-monitoring` project enables you to edit the `user-workload-monitoring-config` `ConfigMap` object. With this role, you can edit the `ConfigMap` object to configure Prometheus, Prometheus Operator, and Thanos Ruler for user-defined workload monitoring.
+|`monitoring-rules-edit` | Users with this role can create, modify, and delete `PrometheusRule` custom resources for a user-defined project. They can also silence alerts.
 
-You can also grant users permission to configure alert routing for user-defined projects:
+|`monitoring-edit` | Users with this role have the same privileges as users with the `monitoring-rules-edit` cluster role. Additionally, users can create, modify, and delete `ServiceMonitor` and `PodMonitor` resources to scrape metrics from services and pods.
```
Comment (juzhao) [inline modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc:26]:
```suggestion
|`monitoring-edit` | Users with this cluster role have the same privileges as users with the `monitoring-rules-edit` cluster role. Additionally, users can create, modify, and delete `ServiceMonitor` and `PodMonitor` resources to scrape metrics from services and pods.
```

```
$ oc get clusterrole monitoring-edit
NAME              CREATED AT
monitoring-edit   2024-05-14T03:41:36Z
```

File: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc [hunk 5]
```diff
@@ -6,22 +6,28 @@
...
 
-* The **alert-routing-edit** cluster role grants a user permission to create, update, and delete `AlertmanagerConfig` custom resources for a project.
+|`user-workload-monitoring-config-edit` | This cluster role is given in the `openshift-user-workload-monitoring` project. Users with this role can edit the `user-workload-monitoring-config` `ConfigMap` object to configure Prometheus, Prometheus Operator, Alertmanager, and Thanos Ruler for user-defined workload monitoring.
```
Comment (juzhao) [inline modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc:28]:
```suggestion
|`user-workload-monitoring-config-edit` | This role is given in the `openshift-user-workload-monitoring` project. Users with this role can edit the `user-workload-monitoring-config` `ConfigMap` object to configure Prometheus, Prometheus Operator, Alertmanager, and Thanos Ruler for user-defined workload monitoring.
```

```
$ oc -n openshift-user-workload-monitoring get role user-workload-monitoring-config-edit
NAME                                   CREATED AT
user-workload-monitoring-config-edit   2024-05-14T03:41:37Z
```

File: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc [hunk 6]
```diff
@@ -6,22 +6,28 @@
...
 
-This section provides details on how to assign these roles by using the {product-title} web console or the CLI.
+|`alert-routing-edit` | Users with this role can create, update, and delete `AlertmanagerConfig` custom resources for a user-defined project.
```
Comment (juzhao) [inline modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc:30]:
```suggestion
|`alert-routing-edit` | Users with this cluster role can create, update, and delete `AlertmanagerConfig` custom resources for a user-defined project.
```
```
$ oc get clusterrole alert-routing-edit
NAME                 CREATED AT
alert-routing-edit   2024-05-14T03:41:37Z
```

File: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc [hunk 7]
```diff
@@ -6,22 +6,28 @@
...
+* To monitor user-defined projects.
+* To configure the components that monitor user-defined projects.
+* To configure alert routing for user-defined projects.
 
-* The *monitoring-rules-edit* cluster role grants a user permission to create, modify, and delete `PrometheusRule` custom resources for a project. It also grants a user the ability to silence alerts.
+You can grant the permissions by assigning one of the following monitoring roles:
 
-* The *monitoring-edit* cluster role grants the same privileges as the `monitoring-rules-edit` cluster role. Additionally, it enables a user to create new scrape targets for services or pods. With this role, you can also create, modify, and delete `ServiceMonitor` and `PodMonitor` resources.
+|===
+|Cluster role name |Description
```
Comment (juzhao) [inline modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc:20]:
```suggestion
|Role name |Description
```
reason, see from https://github.com/openshift/openshift-docs/pull/75852/files#r1599953870, `user-workload-monitoring-config-edit` is role, not cluster role

File: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc (hunk 2 above)
Comment (eromanova97) [reply modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc:24]: Hello @juzhao, I noticed that in the original list we did not include `monitoring-alertmanager-edit` , I could add that role to the table as well, WDYT?

Comment (juzhao) [reply modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc]: @eromanova97 monitoring-alertmanager-edit and cluster-monitoring-view,  thery are used to silence alert, this PR does not contain silence alert. @simonpasquier WDYT?

Comment (simonpasquier) [reply modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc]: monitoring-alertmanager-edit and cluster-monitoring-view roles are specific to platform components while this PR is about user-defined monitoring. We should be more explicit about these roles but probably not in this PR?

Comment (eromanova97) [reply modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc]: Yes, I was thinking about this as well. Maybe for now I will just include the user-defined list as it was intended. We can raise a new issue to create something similar for other roles. :+1: thank you for your inputs!

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

**Comment by JoeAldinger** [comment]
> /lgtm

qe approval

**Pull request merged**